beacon-chain/powchain/service.go

```diff
@@ -725,3 +727,20 @@ func (s *Service) logTillChainStart() {
 		"Time till minimum genesis": time.Duration(secondsLeft) * time.Second,
 	}).Infof("Currently waiting for chainstart")
 }
+
+// cacheHeadersForEth1DataVote makes sure that voting for eth1data after startup utilizes cached headers
+// instead of making multiple RPC requests to the ETH1 endpoint.
+func (s *Service) cacheHeadersForEth1DataVote(lastKnownHeader *gethTypes.Header) {
+	blocksPerVotingPeriod := params.BeaconConfig().EpochsPerEth1VotingPeriod * params.BeaconConfig().SlotsPerEpoch *
+		params.BeaconConfig().SecondsPerSlot / params.BeaconConfig().SecondsPerETH1Block
+
+	end := lastKnownHeader.Number.Uint64() - params.BeaconConfig().Eth1FollowDistance
```

Comment:
Please use the helper to calculate this:
https://github.com/prysmaticlabs/prysm/blob/f31d40cf347d871535625cc7f5a10f16afc72be8/beacon-chain/powchain/service.go#L341